src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -76,103 +110,127 @@
-     * IllegalArgumentException.
-     *
-     * @param longArgName
-     * @return Set<String> of allowable values, or empty set if any value is allowed
-     */
-    public abstract Set<String> getAllowedValuesForDescriptorArgument(String longArgName);
-
-    /**
-     * Called by the command line parser when an argument value from the class specified
-     * by dependentClass has been seen on the command line.
+     * Called by the command line parser when an argument value contained in the class specified
+     * by {@code predecessorClass} has been seen on the command line.
      *
-     * Return true if the argument is allowed (i.e., this name of this class was specified
-     * as a predecessor on the command line) otherwise false.
+     * Return true if the argument is allowed (i.e., this name of {@code predecessorClass} was
+     * specified as a predecessor on the command line), otherwise false.
      *
      * This method can be used by both the command line parser and the descriptor class for
      * determining when to issue error messages for "dangling" arguments (dependent arguments
      * for which a value has been supplied on the command line, but for which the predecessor
-     * argument was not supplied).
+     * plugin was not supplied).
      *
-     * When this method returns "false", the parser will issue an error message if an argument
-     * value in this class has been set on the command line.
+     * When this method returns "false", the parser will issue an error message.
      *
-     * @param dependentClass
+     * @param predecessorClass class which contains the argument that is the subject of the query
      * @return true if the plugin for this class was specified on the command line, or the
      * values in this class may be set byt he user, otherwise false
      */
-    public abstract boolean isDependentArgumentAllowed(Class<?> dependentClass);
+    public abstract boolean isDependentArgumentAllowed(Class<?> predecessorClass);
 
     /**
      * This method is called after all command line arguments have been processed to allow
-     * the descriptor to validate the plugin arguments that have been specified.
+     * the descriptor to validate and resolve the plugin arguments that have been specified, and to
+     * establish any state necessary to implement {@link #getResolvedInstances}.
      *
-     * It is the descriptor's job to contain an argument list which will be populated
-     * by the command line parser with the name of each plugin specified on the command line,
-     * and for each such plugin, to maintain a list of the corresponding instance. This
-     * method gives the descriptor a chance to reduce that list to include only those
-     * instances actually seen on the command line.
+     * It is the descriptor's job to maintain a list of all plugins discovered by the command
+     * line parser (signaled by calls to {@link #createInstanceForPlugin}), and an @Argument
```

Comment:
These requirements should probably be bumped up to the class comment

Reply:
Done above.